### ISSUE TITLE
Add rule to cover case when passing `any` param to function with defined type for the argument 

### ISSUE THREAD

**Issue by deser** [typescript-eslint/typescript-eslint#1982]
I have all `unsafe` rules turned on but they don't warn me about case, when I'm passing `any` to function as argument. Consider example below: 
```TS
export default function () {
    const lala: any = null;

    function kva(columns: string) {
        console.log(columns);
    }

    kva(lala);
}
```

Unsafe rules which are turned on:
```
'@typescript-eslint/no-explicit-any': 1,
'@typescript-eslint/no-unsafe-assignment': 1,
'@typescript-eslint/no-unsafe-call': 1,
'@typescript-eslint/no-unsafe-member-access': 1,
'@typescript-eslint/no-unsafe-return': 1,
```

**Expected Result**
Eslint warns me that passing lala of `any` type to function where this param has not `any` type. 

**Actual Result**
![image](https://user-images.githubusercontent.com/3010698/81147557-34f23e00-8f83-11ea-9d97-b603c940ec3a.png)
(There is no warning)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.31.0` |
| `@typescript-eslint/parser`        | `2.31.0` |
| `TypeScript`                       | `3.8.1` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.3` |
| `npm`                              | `6.14.5` |

**Comment by bradzacher** [comment]
It's on the list of things to do!

See #1968, and comment: https://github.com/typescript-eslint/typescript-eslint/issues/791#issuecomment-622999550
Merging into $791

**Comment by deser** [comment]
Thanks :)
